query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryExecutor.scala

```diff
@@ -253,15 +254,22 @@ class QueryExecutor(httpClient: HttpClient,
               val cache = new Thread {
                 setDaemon(true)
                 setName("Cache thread")
+                val mdc = MDC.getCopyOfContextMap
 
                 override def run(): Unit = {
-                  using(resourceScopeHandle.duplicate()) { handle =>
-                    ready.release()
-                    val isCached = doCache(dataset, unlimitedAnalyses, relimitedAnalyses, context, schema, rollupName, obfuscateId, rowCount, headers, forCache,
-                            handle.get, cacheSession, startWindow.window, endWindow.window, startTimeMs)
-                    if (isCached) { // only do logging if it is not in cache and we do decide to cache.
-                      log.info("Not in cache!")
+                  try {
+                    MDC.setContextMap(mdc)
+                    using(resourceScopeHandle.duplicate()) { handle =>
+                      ready.release()
+                      val isCached = doCache(dataset, unlimitedAnalyses, relimitedAnalyses, context, schema, rollupName, obfuscateId, rowCount, headers, forCache,
+                                             handle.get, cacheSession, startWindow.window, endWindow.window, startTimeMs)
+                      if (isCached) { // only do logging if it is not in cache and we do decide to cache.
+                        log.info("Not in cache!")
+                      }
                     }
+                  } catch {
+                    case e: Throwable =>
+                      log.error("Exception while caching response", e)
```

Comment:
and I guess we're okay with just catching (not throwing it) since its just a caching failure?

Reply:
@rjmac ?

Reply:
Yep, for now.  The main change here is that now we'll log it with appropriate metadata instead of letting the uncaught-exception handler print it to stderr without any context.